src/banjax.cpp

```diff
@@ -254,6 +262,23 @@ Banjax::read_configuration()
 
   //now we can make the filters
   build_filters();
+
+
+  // XXX what should we do if there is no kafka_conf?
+  // * don't make new Kafka* instances (and delete if existing)?
+  //   - useful for testing purposes
+  // * try some default config?
+  if (kafka_consumer == nullptr) {
+    kafka_consumer = std::make_unique<KafkaConsumer>(kafka_conf, this);
+  } else {
+    kafka_consumer->reload_config(kafka_conf, this);
+  }
+
+  // XXX FIXME making a unique_ptr *and* a shared_ptr to this thing...
+  kafka_producer = std::make_unique<KafkaProducer>();
+  challenger->kafka_producer = kafka_producer.get();
+  challenger->kafka_producer->load_config(kafka_conf);
+
```

Comment:
Sorry, but I don't get the difference between handling producer vs consumer here. Why just don't delete/recreate them both from scratch? BTW, what is the difference between `load_config` and `reload_config`?

Reply:
Producer is easier because it doesn't require a (lengthy) graceful shutdown. Consumer needs to shutdown gracefully, so I give it a thread to work in (so i don't block the thread ATS uses to call the config_reload callback).

I guess I can move KafkaProducer's `load_config()` stuff into the initializer if that's better? I think the BajaxFilter-derived classes were all using this two-stage initialization, so I just thought to make it that way. Probably unnecessary, so lemme have a go at it.

Reply:
Sure. Sounds good.